homeassistant/components/homekit/accessories.py

```diff
@@ -1,21 +1,59 @@
 """Extend the basic Accessory and Bridge functions."""
+from datetime import timedelta
+from functools import partial, wraps
+from inspect import getmodule
 import logging
 
 from pyhap.accessory import Accessory, Bridge, Category
 from pyhap.accessory_driver import AccessoryDriver
 
-from homeassistant.helpers.event import async_track_state_change
+from homeassistant.core import callback
+from homeassistant.helpers.event import (
+    async_track_state_change, track_point_in_utc_time)
+from homeassistant.util import dt as dt_util
 
 from .const import (
-    ACCESSORY_MODEL, ACCESSORY_NAME, BRIDGE_MODEL, BRIDGE_NAME,
-    MANUFACTURER, SERV_ACCESSORY_INFO, SERV_BRIDGING_STATE,
+    DEBOUNCE_TIMEOUT, ACCESSORY_MODEL, ACCESSORY_NAME, BRIDGE_MODEL,
+    BRIDGE_NAME, MANUFACTURER, SERV_ACCESSORY_INFO, SERV_BRIDGING_STATE,
     CHAR_MANUFACTURER, CHAR_MODEL, CHAR_NAME, CHAR_SERIAL_NUMBER)
 from .util import (
     show_setup_message, dismiss_setup_message)
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def debounce(func):
+    """Decorator function. Debounce callbacks form HomeKit."""
+    @callback
+    def call_later_listener(hass, acc, value, *args):
+        """Callback listener called from call_later."""
+        nonlocal remove_listener
+        remove_listener = None
+        hass.add_job(func, acc, value)
```

Comment:
You've annotated this method with `@callback`, which means that it will be run inside the event loop, so this should be `hass.async_add_job(func, *lastargs)`